### PR TITLE
Fix Calendar sends duplicate notifications if there is an additional attendee (#21370)

### PR DESCRIPTION
**Pull request by kominoshja**
The bug disclosed on #21370 still occurs on Nextcloud 23. The patch provided by @EhiOnime fixes this issue, but they never made a PR, so it continued to happen.

### CODE REVIEW

**Comment by miaulalala** [comment]
Some tests are failing with these changes, for example:


``` 
There was 1 failure:
149 |  
150 | 1) OCA\DAV\Tests\unit\CalDAV\Reminder\BackendTest::testGetRemindersToProcess
151 | Failed asserting that two arrays are equal.
152 | --- Expected
153 | +++ Actual
154 | @@ @@
155 | 'is_recurrence_exception' => false
156 | 'event_hash' => 'asd123'
157 | 'alarm_hash' => 'asd567'
158 | -    'type' => 'AUDIO'
159 | +    'type' => 'EMAIL'
160 | 'is_relative' => true
161 | 'notification_date' => 123456
162 | 'is_repeat_based' => false
163
``` 

Greatly appreciated if you could take a look at those, but if you're not sure how to fix them, let me know and I'll dive in too. :)

**Comment by kominoshja** [comment]
Thanks for looking into this @miaulalala!

> Greatly appreciated if you could take a look at those, but if you're not sure how to fix them, let me know and I'll dive in too. :)

I'd love to, but I'm not sure how to do it. Do you happen to know what links I need to read so I can figure out how to do this?

**Comment by miaulalala** [comment]
Nmv, looks like it was a one off. :)

**Review by miaulalala — APPROVED** [review]

File: apps/dav/lib/CalDAV/Reminder/Backend.php [hunk 1]
```diff
@@ -68,7 +68,8 @@ public function getRemindersToProcess():array {
 			->from('calendar_reminders', 'cr')
 			->where($query->expr()->lte('cr.notification_date', $query->createNamedParameter($this->timeFactory->getTime())))
 			->join('cr', 'calendarobjects', 'co', $query->expr()->eq('cr.object_id', 'co.id'))
-			->join('cr', 'calendars', 'c', $query->expr()->eq('cr.calendar_id', 'c.id'));
+			->join('cr', 'calendars', 'c', $query->expr()->eq('cr.calendar_id', 'c.id'))
+			->groupBy(['cr.event_hash','cr.type','cr.uid']);
```
Comment (nickvergessen) [inline apps/dav/lib/CalDAV/Reminder/Backend.php:72]: This fails on some DB configurations:
https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_only_full_group_by

Comment (ChristophWurst) [reply apps/dav/lib/CalDAV/Reminder/Backend.php:72]: Select those columns on L67 as well, then you can still run this on the DB

Comment (miaulalala) [reply apps/dav/lib/CalDAV/Reminder/Backend.php:72]: what are you doing here quisof

Comment (nickvergessen) [reply apps/dav/lib/CalDAV/Reminder/Backend.php:72]: > Select those columns on L67 as well

That's not right it. It's the other way around: All items that are selected need to be aggregated.
But I agree, if we previously did not group on db level, just group manually in php.

**Comment by irdiOL** [comment]
This problem persists on NC25. Any plans to merge this patch? Thank you!

**Comment by skjnldsv** [comment]
Fixed already by https://github.com/nextcloud/server/pull/34909

**Pull request closed without merging**